src/components/Articles.jsx

```diff
@@ -5,29 +5,44 @@ import NavArticle from "./NavArticle";
 import ListArticles from "./ListAritcles.jsx";
 
 const Articles = () => {
-
+const { topic } = useParams();
 const [isLoading, setIsLoading] = useState(true);
 const [articles, setArticles] = useState([])
-
-const { topic } = useParams()
+const [params, setParams ] = useState({})
 
 
 useEffect(() => {
     setIsLoading(true);
-    getArticles(topic).then((res) => {
-        setArticles(res)
+    getArticles(params).then(({ articles }) => {
+        setArticles(articles)
         setIsLoading(false);
     })
+}, [params])
+
+useEffect(() => {
+    setParams({ topic })
 }, [topic])
 
+const handleSort = (column) => {
```

Comment:
`const handleSort = (event) => {`
`    const column = event.target.value;`
Then the rest of the function will work